triage/2024-03-19.md

```diff
@@ -0,0 +1,189 @@
+67 artifact comparisons made in total
+
+#### Regressions
+
+Make `DefiningAnchor::Bind` only store the opaque types that may be constrained, instead of the
+current infcx root
+item. [#121796](https://github.com/rust-lang/rust/pull/121796) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=6554a5645a13e4d9331fd028960d69be91d7492d&end=4ccbb7dc95596c7fc0c5756fdc47a17a56c085d3&stat=instructions:u)
+
+|         (instructions:u)          | mean |    range     | count |
+|:---------------------------------:|:----:|:------------:|:-----:|
+|  Regressions ❌ <br /> (primary)   | 0.6% | [0.2%, 1.8%] |  112  |
+| Regressions ❌ <br /> (secondary)  | 0.7% | [0.2%, 1.8%] |  27   |
+|  Improvements ✅ <br /> (primary)  |  -   |      -       |   0   |
+| Improvements ✅ <br /> (secondary) |  -   |      -       |   0   |
+|         All ❌✅ (primary)          | 0.6% | [0.2%, 1.8%] |  112  |
+
+* This is a refactoring of the "old" trait solver that was necessary for follow-up compiler changes.
+
+Rollup of 12 pull
+requests [#122389](https://github.com/rust-lang/rust/pull/122389) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=7de1a1f6db26cf7af43cca74819118428e6317ee&end=6b1e5d9db3b0d569bd847fdcb98c85ee8f7a8ef1&stat=instructions:u)
+
+|         (instructions:u)          | mean |    range     | count |
+|:---------------------------------:|:----:|:------------:|:-----:|
+|  Regressions ❌ <br /> (primary)   | 0.7% | [0.7%, 0.7%] |   3   |
+| Regressions ❌ <br /> (secondary)  | 1.1% | [0.5%, 2.1%] |   3   |
+|  Improvements ✅ <br /> (primary)  |  -   |      -       |   0   |
+| Improvements ✅ <br /> (secondary) |  -   |      -       |   0   |
+|         All ❌✅ (primary)          | 0.7% | [0.7%, 0.7%] |   3   |
+
+* Noise, marked as triaged.
+
+Verify that query keys result in unique dep
+nodes [#122227](https://github.com/rust-lang/rust/pull/122227) [(Comparison Link)](https://perf.rust-lang.org/compare.html?start=e61dcc7a0ac33ef054d76453307124233edcf545&end=d3555f3d8e555ce488bbf8eee5eccdb66a464e14&stat=instructions:u)
+
+|         (instructions:u)          | mean  |     range      | count |
+|:---------------------------------:|:-----:|:--------------:|:-----:|
+|  Regressions ❌ <br /> (primary)   | 2.0%  |  [0.2%, 7.8%]  |  193  |
+| Regressions ❌ <br /> (secondary)  | 2.4%  |  [0.3%, 7.8%]  |  153  |
+|  Improvements ✅ <br /> (primary)  |   -   |       -        |   0   |
+| Improvements ✅ <br /> (secondary) | -1.3% | [-1.3%, -1.3%] |   1   |
+|         All ❌✅ (primary)          | 2.0%  |  [0.2%, 7.8%]  |  193  |
+
+* This PR adds additional verification to the compiler that checks if incremental compilation didn't
+  encounter problems.
+* This result looks quite bad, but that is mostly caused by the configuration of our benchmark
+  suite,
+  which fully enables this verification to make benchmarking more deterministic.
+* Normal builds of user code should only pay 1/32 of the cost here, so it should not be an issue.
```

Comment:
I had the same misconception, but I believe is wrong. My understanding (which could be wrong) is: the checks didn’t change themselves, but new checks were added under the same feature gate, which rustc-perf enables. I don’t believe the checks are done outside of debug-assertions, and the "force incremental verification checks" opt-in, i.e. not present in the 1/32nd of all query calls.

Reply:
Ah, I see, makes sense. So in other words, it shouldn't be a change for normal Rust code.

Reply:
Ah, I see, makes sense. So in other words, it shouldn't be a change for normal Rust code.

Reply:
Ah, I see, makes sense. So in other words, it shouldn't be a change for normal Rust code.

Reply:
I believe so, from what I remember of the PR

Reply:
Ok, reworded.